sdk/azidentity/client_secret_credential.go

```diff
@@ -32,8 +32,10 @@ type ClientSecretCredentialOptions struct {
 	// the application responsible for ensuring the configured authority is valid and trustworthy.
 	DisableInstanceDiscovery bool
 
-	// TokenCachePersistenceOptions enables persistent token caching when not nil.
-	TokenCachePersistenceOptions *TokenCachePersistenceOptions
+	// Cache is a persistent cache the credential will use to store the tokens it acquires, making
```

Comment:
Would this be better served to be on TokenCredentialOptions and the have each token credential include `azidentity.TokenCredentialOptions`.

Avoids needing the same line and comment in each credentialtype

Reply:
It's worth it 😄

Embedding makes literal syntax more awkward e.g.
```go
opts := azidentity.ClientSecretCredentialOptions{
    Cache: c,
}
```
becomes
```go
opts := azidentity.ClientSecretCredentialOptions{
    EmbeddedOptions: azidentity.EmbeddedOptions{
        Cache: c,
    },
}
```

Embedding also adds a layer of indirection in docs. See for example how the embedded `azcore.ClientOptions` appears in [package docs](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#AzurePipelinesCredentialOptions).

I'd also be afraid to add persistent caching to a consolidated options type because it doesn't fit with the other options shared by several credential types and we may in the future add more partly shared options. For example, `OnBehalfOfCredential` doesn't support persistent caching but shares `AdditionallyAllowedTenants` and `DisableInstanceDiscovery` options with several other types that do support persistent caching. So, consolidating and embedding would give us options types having the same field at different levels or options fragmented across a collection of small consolidating types, or some mix of both.